packages/runtime/container-runtime/src/summary/summarizerNode/summarizerNodeWithGc.ts

```diff
@@ -422,22 +413,18 @@ export class SummarizerNodeWithGC extends SummarizerNode implements IRootSummari
 		}
 
 		// In case we have pending summaries on the parent, let's initialize it on the child.
-		if (child.latestSummary !== undefined) {
-			for (const [key, value] of this.pendingSummaries.entries()) {
-				const summaryNodeWithGC = value as SummaryNodeWithGC;
-				if (summaryNodeWithGC.serializedUsedRoutes !== undefined) {
+		if (child.referenceSequenceNumber !== undefined) {
+			for (const [key, pendingSummary] of this.pendingSummaries.entries()) {
+				const pendingSummaryeWithGC = pendingSummary as PendingSummaryInfoWithGC;
```

Comment:
I don't know if this is possible, is there a way to remove this cast? Also, we don't need to do that in this PR